File: src/main/java/cn/dsc/ufocus/param/role/Role.java
package cn.dsc.ufocus.param.role;

import cn.dsc.ufocus.param.user.UserOption;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

/**
 * @author ding.shichen
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Role extends RoleItem {

    /**
     * 最后更新人
     */
    private UserOption latestUpdateUser;

    /**
     * 最后更新时间
     */
    private LocalDateTime updateTime;
}

File: src/main/kotlin/cn/dsc/ufocus/convert/Role.kt
// @author  ding.shichen
// @email   foreverhuiqiao@126.com
// @date    2023-12-11

package cn.dsc.ufocus.convert

import cn.dsc.ufocus.entity.RoleEntity
import cn.dsc.ufocus.param.role.Role
import cn.dsc.ufocus.param.role.RoleItem
import cn.dsc.ufocus.param.role.RoleOption
import cn.dsc.ufocus.param.user.UserOption

fun RoleEntity.toOption() = RoleOption().also {
    it.id = id
    it.chnName = chnName
}

fun RoleEntity.toItem() = RoleItem().also {
    it.id = id
    it.chnName = chnName
    it.createUser = UserOption(createUserId)
    it.createTime = createTime
}

fun RoleEntity.toDetail() = Role().also {
    it.id = id
    it.chnName = chnName
    it.createUser = UserOption(createUserId)
    it.createTime = createTime
    it.latestUpdateUser = UserOption(updateUserId)
    it.updateTime = updateTime
}